include/deal.II/numerics/vector_tools_mean_value.templates.h

```diff
@@ -209,7 +209,9 @@ namespace VectorTools
                         continue;
 
                       // Then adjust its value:
-                      solution(idx) = copy(idx) + constant_adjustment;
+                      solution(idx) =
+                        typename VectorType::value_type(copy(idx)) +
+                        constant_adjustment;
```

Comment:
I don't quite understand this change. `copy` is of type `VectorType`. How could one of its elements be of something other than `VectorType::value_type`?

Reply:
Or, put differently: What is the error message you see here?

Reply:
The type of `copy(idx)` is `internal::VectorReference` and overload resolution for `operator+(internal::VectorReference, const complex<double>)` fails (but `VectorReference has a conversion operator to `PetscScalar`):
```
/tmp/dealii/include/deal.II/numerics/vector_tools_mean_value.templates.h:212:49: error: invalid operands to binary expression ('internal::VectorReference' and 'const typename Vector::value_type' (aka 'const complex<double>'))
                      solution(idx) = copy(idx) + constant_adjustment;
                                      ~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~
/tmp/dealii/build/source/numerics/vector_tools_mean_value.inst:5460:2: note: in instantiation of function template specialization 'dealii::VectorTools::add_constant<dealii::PETScWrappers::MPI::Vector, 1, 2>' requested here
 add_constant<>( PETScWrappers::MPI::Vector  & solution,
 ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/complex:547:1: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('dealii::PETScWrappers::internal::VectorReference' vs. 'double')
operator+(const _Tp& __x, const complex<_Tp>& __y)
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__iterator/move_iterator.h:172:1: note: candidate template ignored: could not match 'move_iterator' against 'complex'
operator+(typename move_iterator<_Iter>::difference_type __n, const move_iterator<_Iter>& __x)
[...]
```

Reply:
Oh, I see. That makes sense, even though it's awkward.